exporters/dsPrometheusExporter/dsPrometheusExporter.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var pingCounter = prometheus.NewCounter(
+	prometheus.CounterOpts{
+		Name: "ping_request_count",
+		Help: "No of request handled by Ping handler",
+	},
+)
+
+func ping(w http.ResponseWriter, req *http.Request) {
+	if req.URL.Path != "/ping" {
+		// This handles unknown endpoints like 404 status code and an empty response body
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+
+	pingCounter.Inc()
```

Comment:
The code as implemented here is demonstrating a self-instrumenting service.
In other words, the code is generating metrics along the way, rather than collecting them when `/metrics` is accessed.

The datasource exporter is going to export metric from another system (the grafana operator) upon accessing the `/metrics` endpoint.

I think that if we implement this sort of dummy exporter now, we will have a big gap between this stage and the one in which we implement the actual exporter logic.

I suggest that we instead implement now a dummy exporter that is more on the path between the dummy service and the intended exporter. 

Some background here: https://prometheus.io/docs/instrumenting/writing_exporters/

There's an interface that such exporter needs to implement. It is explained here: https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#Collector

You can also use the dummy exporter on the ticket as an example.

Reply:
@yftacherzog 
My understanding is: that you recommending a different approach where we create a separate dummy exporter that collects and exports metrics from the dummy service.

Something like a custom Prometheus collector like the `fooCollector` (one you used in your [example code](https://github.com/yftacherzog/prometheus-exporter))

Reply:
The _service_ in the context of this epic _is_ the exporter. It's ultimately going to export metrics it will generate for the Grafana datasource.

We started by creating it as a dummy service in order to be able to create it in as smaller increments as possible. i.e. we create something that can be deployed in Kubernetes -> we make it behave like an exporter -> we make it export Grafana datasource metrics.

All in all, we only create one service -- the exporter.
We're at the point of turning it from a dummy service to a dummy exporter (this ticket and PR).
Next, we'll make it export meaningful stuff.

Reply:
@yftacherzog That makes sense. Thank you for providing a detailed explanation of the ticket's goal.